spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/core/PubSubConfigurationTests.java

```diff
@@ -349,31 +386,29 @@ void testComputeRetryableCodes_returnCustom() {
     PubSubConfiguration pubSubConfiguration = new PubSubConfiguration();
     PubSubConfiguration.Subscriber subscriber = new PubSubConfiguration.Subscriber();
     subscriber.setRetryableCodes(new Code[] {Code.INTERNAL});
-    pubSubConfiguration
-        .getSubscription()
-        .put("projects/projectId/subscriptions/subscription-name", subscriber);
+    pubSubConfiguration.setSubscription(
+        Collections.singletonMap("projects/projectId/subscriptions/subscription-name", subscriber));
+    pubSubConfiguration.initialize("projectId");
 
     assertThat(pubSubConfiguration.computeRetryableCodes("subscription-name", "projectId"))
         .containsExactly(Code.INTERNAL);
   }
 
+  // ATTENTION: this test shows a change in behavior. In the past, specific subscription properties
```

Comment:
To verify, I'm guessing that this change in behavior is in terms of implementation details? The actual behavior of how the properties are picked up still remains the same (i.e. user provided properties come first and then global). Is that an accurate understanding?

Reply:
Right; the difference is that the default properties used to get inserted into the map under multiple subscription keys (for all subscription where there was no dedicated configuration), but now only user-provided properties for a specific subscriptions cause that subscription to be present in the map. Global settings are still returned for subscriptions lacking custom configuration, but they are returned right away, without being placed in the map first.

Reply:
Nice!